pangocffi/attributes.py

```diff
@@ -0,0 +1,478 @@
+from . import (FontDescription, Stretch, Style, Underline, Variant, Weight,
+               ffi, pango)
+
+
+class Attribute:
+    """
+    The :class:`Attributes` — Font and other attributes for annotating text
+    """
+
+    def _init_pointer(self, pointer: ffi.CData):
+        self._pointer = ffi.gc(pointer, pango.pango_attribute_destroy)
+
+    @classmethod
+    def from_pointer(cls, pointer: ffi.CData) -> "Attribute":
+        """
+        Instantiates a :class:`Attribute` from a pointer.
+
+        :return:
+            the Attribute.
+        """
+        if pointer == ffi.NULL:
+            raise ValueError("Null pointer")
+        self = object.__new__(cls)
+        cls._init_pointer(self, pointer)
+        return self
+
+    def get_pointer(self) -> ffi.CData:
+        """
+        Returns the pointer to the context
+
+        :return:
+            the pointer to the context.
+        """
+        return self._pointer
+
+    def __eq__(self, other: "Attribute") -> bool:
+        if isinstance(other, Attribute):
+            return bool(
+                pango.pango_attribute_equal(
+                    self.get_pointer(),
+                    other.get_pointer(),
+                )
+            )
+        return NotImplemented
+
+    @property
+    def start_index(self):
+        return self._start_index
+
+    @start_index.setter
+    def start_index(self, start_index: int):
+        """
+        Set's the Start Index of a Attribute.
+
+        :param start_index:
+            the start index of the range. Should be >=0.
+        """
+        assert start_index >= 0
+        self._start_index = start_index
+        start = ffi.cast("guint", start_index)
+        self._pointer.start_index = start
+
+    @property
+    def end_index(self):
+        return self._end_index
+
+    @end_index.setter
+    def end_index(self, end_index: int):
+        """
+        Set's the End Index of a Attribute.
+
+        :param end_index:
+            end index of the range. The character at this index
+            is not included in the range.
+        """
+        assert end_index >= 0
+        self._start_index = end_index
+        end = ffi.cast("guint", end_index)
+        self._pointer.start_index = end
+
+    @classmethod
+    def from_language(
+        cls, language, start_index: int = 0, end_index: int = 1
+    ) -> "Attribute":
+        """
+        API not implemented
+        """
+        temp = cls.from_pointer(pango.pango_attr_language_new(language))
+        temp.start_index = start_index
+        temp.end_index = end_index
+        return temp
+
+    @classmethod
+    def from_family(
```

Comment:
@leifgehrmann can you suggest me a better way that this?

Reply:
:+1: to this.

Reply:
Note: I copied the end_index from https://developer.gnome.org/pygtk/stable/class-pangoattribute.html idk why they defined so. But there could be a reason for it.

Reply:
Thanks for referencing link, I hadn't seen it!

In that case, I think it's more reasonable for it to remain as 1 then. I don't want to diverge too much from their API and cause confusion.